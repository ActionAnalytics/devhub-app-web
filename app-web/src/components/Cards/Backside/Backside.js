/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import React from 'react';
import PropTypes from 'prop-types';
import shortid from 'shortid';
import Card from '../../../hoc/Card';
import ActionGroup from './ActionGroup/ActionGroup';

const Backside = ({ actionGroups }) => (
  <Card>
    {actionGroups.map(ag => (
      <ActionGroup title={ag.title} actions={ag.actions} key={shortid.generate()} />
    ))}
  </Card>
);

Backside.propTypes = {
  actionGroups: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      actions: PropTypes.array.isRequired,
    }),
  ).isRequired,
};

export default Backside;
