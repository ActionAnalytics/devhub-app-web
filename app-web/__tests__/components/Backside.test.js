import React from 'react';
import { shallow } from 'enzyme';
import Backside from '../../src/components/Cards/Backside/Backside';

describe('Backside Component (card)', () => {
  it('matches snapshot', () => {
    const props = {
      actionGroups: [
        {
          title: 'Actions',
          actions: [
            {
              icon: <i>Icon</i>,
              actionName: 'foo',
              actionValue: 'bar',
            },
          ],
        },
      ],
    };

    const wrapper = shallow(<Backside actionGroups={props.actionGroups} />);
    expect(wrapper).toMatchSnapshot();
  });
});
