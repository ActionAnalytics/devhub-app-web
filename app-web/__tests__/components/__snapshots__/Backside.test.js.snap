// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backside Component (card) matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Backside
    actionGroups={
      Array [
        Object {
          "actions": Array [
            Object {
              "actionName": "foo",
              "actionValue": "bar",
              "icon": <i>
                Icon
              </i>,
            },
          ],
          "title": "Actions",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "author": "",
      "children": Array [
        <ActionGroup
          actions={
            Array [
              Object {
                "actionName": "foo",
                "actionValue": "bar",
                "icon": <i>
                  Icon
                </i>,
              },
            ]
          }
          title="Actions"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "actions": Array [
            Object {
              "actionName": "foo",
              "actionValue": "bar",
              "icon": <i>
                Icon
              </i>,
            },
          ],
          "title": "Actions",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "author": "",
        "children": Array [
          <ActionGroup
            actions={
              Array [
                Object {
                  "actionName": "foo",
                  "actionValue": "bar",
                  "icon": <i>
                    Icon
                  </i>,
                },
              ]
            }
            title="Actions"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "actions": Array [
              Object {
                "actionName": "foo",
                "actionValue": "bar",
                "icon": <i>
                  Icon
                </i>,
              },
            ],
            "title": "Actions",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
